src/dependency-manager/manager.ts

```diff
@@ -122,6 +122,24 @@ export default abstract class DependencyManager {
           graph.addEdge(edge);
         }
       }
+
+      const database_regex = new RegExp(`\\\${{\\s*(dependencies\\.(?<dependency_name>${ComponentSlugUtils.RegexBase})\\.)?databases\\.(?<database_name>${Slugs.ArchitectSlugRegexBase})\\.(?<database_key>${Slugs.ArchitectSlugRegexBase})`, 'g');
+      while ((matches = database_regex.exec(resource_string)) !== null) {
+        if (!matches.groups) continue;
+
+        const { dependency_name, database_name } = matches.groups;
+
+        const dependency = component_configs.find(c => c.name === dependency_name) || component;
+
+        const to = buildNodeRef(dependency, 'services', `${database_name}-db`);
```

Comment:
This is going to determine how we handle [ResourceTypes](https://github.com/architect-team/architect-cli/blob/91441202a9177e4df32f94723419d6845ef16ffd/src/dependency-manager/spec/utils/slugs.ts#L142), and ultimately how we will handle other entities going forward. 

I like the idea of adding databases to `ResourceType` since those are one of the more important entities that people are concerned about in their applications. This also opens us up to other resources - I can see managed cloud resources as `cloud` ResourceType. 

Also, I think we should drop the `-db` from the name and just keep it as ${database_name}

Reply:
This was just an oversight on my part. I think what TJ wrote is right for the node ref.